### DISCUSSION TITLE
Complains about slider bars in the editor

### DISCUSSION THREAD

**Discussion by cihe13375** [ppy/osu#30032] [Q&A]
This is indeed several issues but I guess it's better to combine them into one discussion.

To be clear, for "slider bar" I mean the green thing in the image below
![aaaa](https://github.com/user-attachments/assets/c853f9a1-7177-42b6-b087-2a41491ad7a8)

I want to complain three things about this slider bar, in decreasing importance imo:
1) for some reason the slider bar has a "trigger threshold", so if the amount of change is not big enough it does not react at all. See the video below, note that the angle is still at 0 and then suddenly jumps to 26, and again jumps from 29 to 55. The current design makes it very difficult to adjust the number by small amount (<30 degrees in this case), which is not a rare operation.

https://github.com/user-attachments/assets/29db6bde-367e-452c-a455-39da2c5c6420

2) in stable one could select the slider bar and then use arrow keys to adjust the number by small increments (1 degree for angle, 0.01x for scaling), see the video below:

https://github.com/user-attachments/assets/3b76cf36-959f-4f4b-882f-c033081c43ff

 It kinda works in lazer but is unintuitive (one need to first deselect the input box, and there is no sign of the slider bar being "selected"). More importantly, the arrow keys in lazer change the numbers in large increments (36 degrees for angle, and some random numbers for scaling), making it not very useful unless you really want to e.g. rotate by exactly 36n degrees (see the second half of the first video).

3) In stable the slider bar has ticks and the handle is small, so it's easy to build a correspondence between the numbers and the location on the bar. Lazer slider bars don't have ticks and the handle is huge, so it basically relies on guessing to know e.g. how much to drag for changing the angle by 50 degrees, or where to click to set the angle to ~100 degrees. Also in lazer one can click on the empty region of the silder bar to immediately direct to there (good), but the handle is covering too much space, making the feature less useful than it could be (e.g. for the angle dialog the handle covers a large span of about -140 to +140 degrees)

**Comment by peppy** [comment]
The minimum drag before the bar starts moving is defined at a framework level. It's usually a sane value, but in this case I'd maybe argue the slider bars you present are just too chubby and need to be longer to work correctly.

I agree they should have visible ticks.
